gapis/api/vulkan/frame_loop.go

```diff
@@ -626,10 +671,67 @@ func (f *frameLoop) resetBuffers(ctx context.Context, stateBuilder *stateBuilder
 		log.D(ctx, "Reset buffer [%v] with buffer [%v] succeed", dst, src)
 	}
 
-	stateBuilder.scratchRes.Free(stateBuilder)
 	return nil
 }
 
+func (f *frameLoop) recreateDestroyedBuffer(ctx context.Context, stateBuilder *stateBuilder, buffer BufferObjectʳ) {
+	memReq := NewVkMemoryRequirements(stateBuilder.ta,
+		buffer.MemoryRequirements().Size(), buffer.MemoryRequirements().Alignment(), buffer.MemoryRequirements().MemoryTypeBits())
+
+	createWithMemReq := stateBuilder.cb.VkCreateBuffer(
+		buffer.Device(),
+		stateBuilder.MustAllocReadData(
+			NewVkBufferCreateInfo(stateBuilder.ta,
+				VkStructureType_VK_STRUCTURE_TYPE_BUFFER_CREATE_INFO, // sType
+				0,                           // pNext
+				buffer.Info().CreateFlags(), // flags
+				buffer.Info().Size(),        // size
+				VkBufferUsageFlags(uint32(buffer.Info().Usage())|uint32(VkBufferUsageFlagBits_VK_BUFFER_USAGE_TRANSFER_DST_BIT|VkBufferUsageFlagBits_VK_BUFFER_USAGE_TRANSFER_SRC_BIT)), // usage
+				buffer.Info().SharingMode(),                      // sharingMode
+				uint32(buffer.Info().QueueFamilyIndices().Len()), // queueFamilyIndexCount
+				NewU32ᶜᵖ(stateBuilder.MustUnpackReadMap(buffer.Info().QueueFamilyIndices().All()).Ptr()), // pQueueFamilyIndices
+			)).Ptr(),
+		memory.Nullptr,
+		stateBuilder.MustAllocWriteData(buffer.VulkanHandle()).Ptr(),
+		VkResult_VK_SUCCESS,
+	)
+	createWithMemReq.Extras().Add(memReq)
+	stateBuilder.write(createWithMemReq)
+	stateBuilder.write(stateBuilder.cb.VkGetBufferMemoryRequirements(
+		buffer.Device(),
+		buffer.VulkanHandle(),
+		stateBuilder.MustAllocWriteData(memReq).Ptr(),
+	))
+
+	mem := buffer.Memory()
```

Comment:
I think this might be wrong.
This is recreating the entire device memory object, regardless of it is actually deleted or not. I think that is incorrect.

Reply:
We should have a separate pass that re-creates device mmory as needed, (that happens before buffers/images)

Reply:
thanks Andrew for the review, I've create another CL the deal with the device memory. After that I'll update this part of the CL. Change the CL for WIP  for now.

Reply:
Just added code to handle the device memory free/allocate separately. Tested with the sample which free/re-allocate vertex buffer and the dump images are as expected.